packages/foam-vscode/src/features/create-from-template.ts

```diff
@@ -153,12 +174,73 @@ export function substituteFoamVariables(
   return templateText;
 }
 
+function sortTemplatesMetadata(templatesMetadata: object[]) {
+  // Sort by name's existence, name, then path
+  return templatesMetadata.sort((t1, t2) => {
+    if (t1['name'] === undefined && t2['name'] !== undefined) {
+      return 1;
+    }
+
+    if (t1['name'] !== undefined && t2['name'] === undefined) {
+      return -1;
+    }
+
+    if (t1['name'] > t2['name']) {
```

Comment:
minor: I wonder if this whole section could be replaced by

```
return t1['name'].localeCompare(t2['name']) 
  || t1['templatePath'].localeCompare(t2['templatePath'])
```

?

(haven't tested it, just a hunch)

Reply:
🎉 That allows it to shrink a bit. We still have to handle the cases where `name` is `undefined`.

I've made the changes. I'm happy to be shown even more compact solutions.

Reply:
yeah I was thinking of some defaults, but I couldn't find something that wouldn't make things more obscure, so I am happy to keep it the way it is